### ISSUE TITLE
Use cysignals

### ISSUE THREAD

**Issue by isuruf** [symengine/symengine.py#56]
See https://github.com/sagemath/cysignals

**Comment by certik** [comment]
Yes, I asked if they could change the license from GPL to LGPL and they did, so we could use it.
